R/cast.R

```diff
@@ -62,11 +63,16 @@
 #'
 #' # Cast to common type
 #' vec_cast_common(factor("a"), factor(c("a", "b")))
-vec_cast <- function(x, to, ..., x_arg = "", to_arg = "") {
+vec_cast <- function(x,
+                     to,
+                     ...,
+                     x_arg = "",
+                     to_arg = "",
+                     call = caller_env()) {
   if (!missing(...)) {
     check_ptype2_dots_empty(...)
   }
-  return(.Call(vctrs_cast, x, to, x_arg, to_arg))
+  return(.Call(ffi_cast, x, to, x_arg, to_arg, environment()))
```

Comment:
Oh I may have answered my question about `call` vs `frame`. I was going to ask if you should be passing `call` through rather than `environment()`, but is the idea that `environment()` is the "frame" and you can look up the call from the `call` object that is present in that frame?

Reply:
yup in this case I'm using `frame` to create a `struct r_lazy` for `call`.